### PR TITLE
Feat/thirdauth

### PR DESCRIPTION
**Pull request by AngheloZambrana**

### CODE REVIEW

File: apps/UserManagement/Backend/Controllers/TokenController.cs [hunk 1]
```diff
@@ -25,10 +25,11 @@ public Task<ActionResult<UserFullInfoDTO>> PostToken(CreateTokenDTO sessionPostD
     [HttpGet("GetCookie/{userId}")]
     public ActionResult<bool> GetCookie(Guid userId)
     {
+        Console.WriteLine("fghsjkdfg");
```
Comment (MayerliSantander) [inline apps/UserManagement/Backend/Controllers/TokenController.cs]: Remove unnecessary console writeline

File: apps/UserManagement/Backend/DTOs/WithID/SubsidiaryDTO.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+public class SubsidiaryDTO
+{
+    public Guid subsidiaryId { get; set; }
+    public decimal latitude { get; set; }
+    public decimal longitude { get; set; }
+    public string name { get; set; }  = string.Empty;
+    public string type { get; set; }  = string.Empty;
+    public Guid companyId { get; set; }
+}
```
Comment (MayerliSantander) [inline apps/UserManagement/Backend/DTOs/WithID/SubsidiaryDTO.cs]: Add endline

File: apps/UserManagement/Backend/DTOs/WithID/SubsidiaryUsersDTO.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+namespace Backend.DTOs.WithID;
+
+public class SubsidiaryUsersDTO
+{
+    public Guid userId { get; set; }
+    public Guid subsidiaryId { get; set; }
+}
```
Comment (MayerliSantander) [inline apps/UserManagement/Backend/DTOs/WithID/SubsidiaryUsersDTO.cs]: Add endline

File: apps/UserManagement/Backend/DTOs/WithoutID/UserBySubsidiaryDTO.cs [hunk 4]
```diff
@@ -3,5 +3,5 @@ namespace Backend.DTOs.WithoutID;
 public class UserBySubsidiaryDTO
 {
     public Guid SubsidiaryId { get; set; }
-    public string Email { get; set; }
+    public string Email { get; set; } = string.Empty;
 }
```
Comment (MayerliSantander) [inline apps/UserManagement/Backend/DTOs/WithoutID/UserBySubsidiaryDTO.cs]: Add endline

File: apps/UserManagement/Backend/Services/ProductAPIService.cs [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        response.EnsureSuccessStatusCode();
+        string content = await response.Content.ReadAsStringAsync();
+        var jsonContent = JsonSerializer.Deserialize<List<SubsidiaryUsersDTO>>(content);
+        if (jsonContent?.Count > 0)
+        {
+            return jsonContent.Select(dto => dto.subsidiaryId).ToList();
+        }
+        return null;
+    }
+}
```
Comment (MayerliSantander) [inline apps/UserManagement/Backend/Services/ProductAPIService.cs]: Add endline

File: apps/UserManagement/Backend/Services/ServiceInterfaces/IProductAPIService.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+namespace Backend.Services.ServiceInterfaces;
+
+public interface IProductAPIService
+{
+    public Task<List<SubsidiaryUsersDTO>?> GetUsersBySubsidiaryId(Guid id);
+    public Task<List<SubsidiaryUsersDTO>?> GetUsersByCompanyId(Guid id);
+    public Task<Guid?> GetCompanyIdByUserId(Guid userId);
+    public Task<List<Guid>?> GetSubsidiariesIdsByUserId(Guid userId);
+}
```
Comment (MayerliSantander) [inline apps/UserManagement/Backend/Services/ServiceInterfaces/IProductAPIService.cs]: Add endline

**Review by MayerliSantander — CHANGES_REQUESTED** [review]

**Review by DeidamiaFuentes — APPROVED** [review]

**Review by MayerliSantander — APPROVED** [review]

**Pull request merged**